commit 941f02ede63baa46f93ed8abccebe76fb29c0789
Author: Lars Knoll <lars@trolltech.com>
Date:   Wed Nov 8 12:17:41 2006 -0800

    Don't hold the display lock around callbacks to the application.
    
    This fixes an XCB locking assertion failure, particularly with emacs.

diff --git a/src/FilterEv.c b/src/FilterEv.c
index 0117291..9fa1887 100644
--- a/src/FilterEv.c
+++ b/src/FilterEv.c
@@ -96,9 +96,9 @@ XFilterEvent(ev, window)
 	if (win == p->window) {
 	    if ((mask & p->event_mask) ||
 		(ev->type >= p->start_type && ev->type <= p->end_type)) {
+		UnlockDisplay(ev->xany.display);
 		ret = (*(p->filter))(ev->xany.display, p->window, ev,
 				      p->client_data);
-		UnlockDisplay(ev->xany.display);
 		return(ret);
 	    }
 	}
